### ISSUE TITLE
enable_ansi_support(): error LNK2019: unresolved external symbol

### ISSUE THREAD

**Issue by iology** [ogham/rust-ansi-term#31]
Compiling failed on AppVeyor CI.

See the logs of the failed builds: https://ci.appveyor.com/project/sharkdp/fd/build/1.0.197

**Comment by volks73** [comment]
I would like to add that I have also experienced the LNK2019 error. I was using the stable-i686-pc-windows-msvc toolchain from the [Developer Prompt](https://docs.microsoft.com/en-us/dotnet/framework/tools/developer-command-prompt-for-vs) installed with the [C++ Built Tools 2017](http://landinghub.visualstudio.com/visual-cpp-build-tools) configured for the x86 environment. When I try to build my project using the stable-x86_64-pc-windows-msvc toolchain from the Developer Prompt configurd for the x64 environment, everything builds successfully. 

Here is a snippet of the error message that displays the linker error.

```txt
note: libansi_term-90d9c9360c432cc8.rlib(ansi_term-90d9c9360c432cc8.ansi_term15.rust-cgu.o) : 
error LNK2019: unresolved external symbol __imp__GetStdHandle referenced in function 
__ZN9ansi_term7windows19enable_ansi_support17h52d3ee1de6502c43E
          libansi_term-90d9c9360c432cc8.rlib(ansi_term-90d9c9360c432cc8.ansi_term15.rust-cgu.o) : 
error LNK2019: unresolved external symbol __imp__GetLastError referenced in function 
__ZN9ansi_term7windows19enable_ansi_support17h52d3ee1de6502c43E
          libansi_term-90d9c9360c432cc8.rlib(ansi_term-90d9c9360c432cc8.ansi_term15.rust-cgu.o) : 
error LNK2019: unresolved external symbol __imp__SetConsoleMode referenced in function 
__ZN9ansi_term7windows19enable_ansi_support17h52d3ee1de6502c43E
```

When I remove the `ansi_term` crate from the project, the build is successful with the stable-i686-pc-windows-msvc toolchain. It appears the problem is related to enabling ansi support and the i686 toolchain. I am not sure how to go about resolving the problem, but I am happy to help in any way I can.

**Comment by retep998** [comment]
Your functions have the wrong signatures. I personally recommend getting your bindings from `winapi` and not trying to bind them yourself.

**Comment by pravic** [comment]
Funny enough that it is reproduced only with release build. At least in Aaronepower/tokei#173.

**Comment by glebpom** [comment]
There are already two pull-requests fixing this issue.  #33 by @volks73  and #35 by me. I think this bug is pretty critical because `clap` relies on this crate, so a lot of software fails to compile for 
 32bit windows platform

**Comment by volks73** [comment]
I agree with @retep998 and @glebpm that using the winapi crate should be preferred, so please use the #35 pull request to resolve this issue.

**Comment by pravic** [comment]
It looks like @ogham is unavailable. 

What Rust community does in such situations?

**Comment by glebpom** [comment]
From [crates policies](https://crates.io/policies):

If someone wants to take over a package, and the previous owner agrees, the existing maintainer can add them as an owner, and the new maintainer can remove them. If necessary, the team may reach out to inactive maintainers and help mediate the process of ownership transfer.

**Comment by pravic** [comment]
Oh my: https://crates.io/crates/ansi_term/reverse_dependencies

I've sent a letter to @ogham but he didn't respond. However he's quite active on SO still, so no idea why this project is unmaintained so long.

**Comment by ogham** [comment]
Hi everyone,

I'm really sorry this took me an age to merge -- I'll try to be a better steward of my crate in the future.

Version v0.11 was published just now with the PR included.

**Comment by sharkdp** [comment]
@ogham Thank you very much!

> I'm really sorry this took me an age to merge -- I'll try to be a better steward of my crate in the future.

Don't let anyone stress you out. This is an open source project, after all.

**Comment by rivy** [comment]
Should this be closed now?

**Comment by iology** [comment]
Not sure. I don't have a Windows machine or Rust project with AppVeyor CI enabled for testing at the moment. :wink:

**Comment by rivy** [comment]
Hmm,  I think I can help with that ... [PR#46](https://github.com/ogham/rust-ansi-term/issues/46).
🙂
